pkg/admission/clusterworkspacetypeexists/admission.go

```diff
@@ -55,8 +55,8 @@ func Register(plugins *admission.Plugins) {
 				Handler:          admission.NewHandler(admission.Create, admission.Update),
 				createAuthorizer: delegated.NewDelegatedAuthorizer,
 			}
-			plugin.transitiveTypeResolver = transitiveTypeResolver{
-				getter: func(cluster logicalcluster.Name, name string) (*tenancyv1alpha1.ClusterWorkspaceType, error) {
+			plugin.transitiveTypeResolver = TransitiveTypeResolver{
```

Comment:
Nit: might want this to be an interface & guarded by a constructor, since the zero value is not a valid thing

Reply:
interfaces should be defined where they are used, not where the type is defined. but we can unexport the type and add a constructor if you want?

Reply:
Sure, `func NewTransitiveTypeResolver(getter func(cluster logicalcluster.Name, name string) (*tenancyv1alpha1.ClusterWorkspaceType, error)) *transitiveTypeResolver` would be fine